app/uk/gov/hmrc/nationalinsurancerecord/controllers/NationalInsuranceRecordController.scala

```diff
@@ -38,7 +38,7 @@ trait NationalInsuranceRecordController extends BaseController with HeaderValida
       content,
       selfLink,
       Some(Vector("taxYears" -> years.map(taxYear => halResourceSelfLink(Json.toJson(taxYear),
-        nationalInsuranceTaxYearHref(nino, taxYear.taxYear))).toVector))
+        nationalInsuranceTaxYearHref(nino, TaxYear(taxYear.taxYear)))).toVector))
```

Comment:
I think you want it to accept TaxYear so it's consistent everywhere? You can have `nationalInsuranceTaxYearHref` continue to accept a TaxYear object, but adjust the `TaxYearSummary` case class to accept a TaxYear rather than a String. Or I'm missing something :)